spdx-utils/src/main/kotlin/SpdxLicenseMapping.kt

```diff
@@ -19,337 +19,331 @@
+            "The JSON License" to JSON.toExpression(),
+            "The MIT" to MIT.toExpression(),
+            "The MIT License" to MIT.toExpression(),
+            "The MIT License (MIT)" to MIT.toExpression(),
+            "The MIT License(MIT)" to MIT.toExpression(),
+            "The MIT license" to MIT.toExpression(),
+            "The New BSD License" to BSD_3_CLAUSE.toExpression(),
+            "The PostgreSQL License" to POSTGRESQL.toExpression(),
+            "The SAX License" to SAX_PD.toExpression(),
+            "The Unlicense" to UNLICENSE.toExpression(),
+            "The W3C License" to W3C.toExpression(),
+            "The W3C Software License" to W3C.toExpression(),
+            "Three-clause BSD-style" to BSD_3_CLAUSE.toExpression(),
+            "Two-clause BSD-style license" to BSD_2_CLAUSE.toExpression(),
+            "UNLICENSE" to UNLICENSE.toExpression(),
+            "UNLICENSED" to UNLICENSE.toExpression(),
+            "Unicode/ICU License" to ICU.toExpression(),
+            "ZPL 2.1" to ZPL_2_1.toExpression(),
+            "Zope Public" to ZPL_2_1.toExpression(),
+            "apache-2.0" to APACHE_2_0.toExpression(),
+            "boost" to BSL_1_0.toExpression(),
+            "http://ant-contrib.sourceforge.net/tasks/LICENSE.txt" to APACHE_1_1.toExpression(),
+            "http://opensource.org/licenses/MIT" to MIT.toExpression(),
+            "http://svnkit.com/license.html" to TMATE.toExpression(),
+            "http://www.apache.org/licenses/LICENSE-2.0" to APACHE_2_0.toExpression(),
+            "http://www.apache.org/licenses/LICENSE-2.0.txt" to APACHE_2_0.toExpression(),
+            "http://www.gnu.org/copyleft/lesser.html" to LGPL_3_0.toExpression(),
+            "lgpl" to LGPL_2_1.toExpression(),
+            "new BSD" to BSD_3_CLAUSE.toExpression(),
             "public domain, Python, 2-Clause BSD, GPL 3 (see COPYING.txt)"
-                    to enumSetOf(SpdxLicense.PYTHON_2_0, SpdxLicense.BSD_2_CLAUSE, SpdxLicense.GPL_3_0),
-            "ruby" to enumSetOf(SpdxLicense.RUBY),
-            "the Apache License, ASL Version 2.0" to enumSetOf(SpdxLicense.APACHE_2_0),
-            "wtfpl" to enumSetOf(SpdxLicense.WTFPL),
-            "zlib" to enumSetOf(SpdxLicense.ZLIB)
+                    to (((licenseRef("scancode-public-domain-disclaimer") and PYTHON_2_0.toExpression())
+                    and BSD_2_CLAUSE.toExpression()) and GPL_3_0.toExpression()),
+            "ruby" to RUBY.toExpression(),
+            "the Apache License, ASL Version 2.0" to APACHE_2_0.toExpression(),
+            "wtfpl" to WTFPL.toExpression(),
+            "zlib" to ZLIB.toExpression()
     )
 
-    fun map(license: String) =
-            mapping.getOrDefault(license, enumSetOf()).apply {
-                // Also add any direct match for an SPDX ID.
-                SpdxLicense.forId(license)?.let { add(it) }
-            }
+    private fun licenseRef(id: String) = SpdxLicenseReferenceExpression("LicenseRef-$id")
```

Comment:
Should we maybe add a `namespace` parameter which defaults to "scancode" and omit the namespace from the passed `id` argument then? That would avoid us to hard-code "scancode" in too many places. I believe we yet need to think of how we can use LicenseRefs but still be somewhat scanner-independent. In the end, we probably should only emit "ORT"-namespaced LicenseRefs and maintain another mapping to scanner-specific LicenseRefs.

Reply:
So far the function is only used once, I would wait for a few more samples before changing it, to see what we need.